low-level/sensors/impl/generic/node_hw.py

```diff
@@ -958,19 +959,30 @@ def _parse_fan_info(self, index, date, _time, device_id, sensor_id,\
         original_event = event
         original_status = status
 
+        sensor_name = self.sensor_id_map[self.TYPE_FAN][sensor_id]
+        impact = "NA"
+        recommendation = None
+        description = ""
+        state = "high" if "going high" in event.lower() else "low"
         if status.lower() == "deasserted" and event.lower() == "fully redundant":
             alert_type = "fault"
+            description = f"{sensor_name} sensor is completely redundant"
         elif status.lower() == "asserted" and event.lower() == "fully redundant":
             alert_type = "fault_resolved"
+            description = f"{sensor_name} sensor has recovered from 'fault' state."
         elif "going high" in event.lower() or "going low" in event.lower():
```

Comment:
Also rather than having `if-else` conditions to decide description etc. I think we should have some structured hard code as I tried to create for Discrete Sensors like Disk and PSU, we can use the same or have some better structure for Threshold-based sensors since they have the same events for all the threshold-based sensors like temp, voltage, current, etc.

Reply:
> Also rather than having if-else conditions to decide description etc.

This code was already in-place, if-else condition is to decide alert type based on event
I only introduced description field

Reply:
@sumedhak27  addressed your comment.
I have aligned the changes for Fan module similar to Disk/PSU